### PR TITLE
[feature] Add Hunyuan-GameCraft model support

### PR DESCRIPTION
**Pull request by MihirJagtap**
## Add Hunyuan-GameCraft model support

Add support for **Hunyuan-GameCraft** video generation with camera pose conditioning via Plücker coordinates (T2V and I2V).

### Components added

- **CameraNet** (`fastvideo/models/cameranet.py`): Encodes camera pose into Plücker coordinates for conditioning.
- **HunyuanGameCraftTransformer3DModel**: New 3D DiT integrating CameraNet; 33-channel input (noise + gt_latents + mask) for I2V.
- **GameCraftFinalLayer**: FinalLayer with correct shift/scale order for parity with the official implementation.
- **GameCraft VAE**: Dedicated VAE with `mid_block_causal_attn`, correct `scaling_factor`, and optional tiling/upcast for fp16/bf16.
- **Model and pipeline configs**: GameCraft model/pipeline/sample configs and `param_names_mapping` for weight loading.
- **Pipeline stages**: `GameCraftDenoisingStage` (33-ch concat, CFG, camera_states; I2V ref-latent injection every step) and `GameCraftImageVAEEncodingStage` (reference image → gt_latents + conditioning_mask).
- **Weight conversion scripts**: Convert official PyTorch checkpoints to Diffusers `safetensors` (DiT, VAE, full pipeline, LLaVA text encoder extraction).
- **Numerical parity tests**: Local tests for transformer, VAE, text encoders, and pipeline vs official (RoPE and FinalLayer behavior).
- **SSIM regression tests**: T2V and I2V in `fastvideo/tests/ssim/test_gamecraft_similarity.py` (480p, 20 steps); references under `<GPU>_reference_videos/HunyuanGameCraft-T2V` and `HunyuanGameCraft-I2V`.

### Implementation notes

- **RoPE**: In `get_1d_rotary_pos_embed`, frequency expansion uses `repeat_interleave` (not `torch.cat`) so cos/sin pairing matches the official model.
- **FinalLayer**: GameCraft uses normalize → scale → shift: `x = norm(x) * (1 + scale) + shift`.
- **I2V**: Reference latent is injected into conditioned frames at every denoising step; image encoding stage builds gt_latents and conditioning_mask from the reference image.

### Inference

- **Example**: `examples/inference/basic/basic_gamecraft.py` (T2V/I2V with camera trajectory from official GameCraft helpers).

### CODE REVIEW

File: examples/inference/basic/basic_gamecraft.py
Comment (Eigensystem) [inline examples/inference/basic/basic_gamecraft.py:1]: Why don't we use `StreamingVideoGenerator` and `step` here?

File: fastvideo/configs/models/dits/__init__.py [hunk 1]
```diff
@@ -10,6 +11,9 @@
 from fastvideo.configs.models.dits.hyworld import HYWorldConfig
 
 __all__ = [
+    "HunyuanVideoConfig", "HunyuanVideo15Config", "HunyuanGameCraftConfig",
```
Comment (Eigensystem) [inline fastvideo/configs/models/dits/__init__.py:13]: Remove duplicates plz

File: fastvideo/models/cameranet.py
Comment (Eigensystem) [inline fastvideo/models/cameranet.py:1]: Move to `fastvideo/models/dit/hunyuangamecraft/` or delete it if it's a duplicate.

Comment (MihirJagtap) [reply fastvideo/models/cameranet.py:1]: deleted. It was duplicate. cameranet is in fastvideo/models/dits/hunyuangamecraft.py

File: fastvideo/configs/models/dits/__init__.py (hunk 1 above)
Comment (MihirJagtap) [reply fastvideo/configs/models/dits/__init__.py:13]: removed. thanks for pointing that out!

File: fastvideo/configs/pipelines/__init__.py [hunk 2]
```diff
@@ -15,6 +16,12 @@
                                              WanT2V480PConfig, WanT2V720PConfig)
 
 __all__ = [
+    "HunyuanConfig", "FastHunyuanConfig", "HunyuanGameCraftPipelineConfig",
```
Comment (Eigensystem) [inline fastvideo/configs/pipelines/__init__.py:17]: same here

**Review by Eigensystem — COMMENTED** [review]
plz fix pre-commit errors

File: examples/inference/basic/basic_gamecraft.py
Comment (MihirJagtap) [reply examples/inference/basic/basic_gamecraft.py:1]: StreamingVideoGenerator only works with pipelines that implement the streaming interface: streaming_reset(), streaming_step(), and streaming_clear().
The GameCraft pipeline does not define these.

The streaming step() is built for MatrixGame’s per-step controls:
step(keyboard_cond, mouse_cond)
GameCraft uses one camera_states tensor (Plücker coords) for the whole clip. There is no “one action per step” API; the full trajectory is fixed before generation.

So in examples/inference/basic/basic_gamecraft.py we use normal VideoGenerator and generate_video() because GameCraft generate full clip in one go. It has camera path already computed, not a step by step streaming loop

File: examples/inference/basic/basic_hunyuangamecraft.py [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+    Returns:
+        camera_states: [1, num_frames, 6, height, width] Plücker embeddings
+    """
+    # Add official GameCraft to path
+    official_path = repo_root / "Hunyuan-GameCraft-1.0"
+    if official_path.exists() and str(official_path) not in sys.path:
+        sys.path.insert(0, str(official_path))
+    
+    try:
+        from hymm_sp.sample_inference import (
```
Comment (Eigensystem) [inline examples/inference/basic/basic_hunyuangamecraft.py]: Do we still rely on hunyuan gamecraft official repo now? Maybe we should move these components to our repo if needed.

**Comment by Eigensystem** [comment]
Hi @MihirJagtap. Could you please merge this branch: [hunyuan-gamecraft-v2](https://github.com/hao-ai-lab/FastVideo/tree/hunyuan-gamecraft-v2). I fixed some runtime errors but had no access to your repo for committing to the PR.
